docs/protocol_overview.md

```diff
@@ -0,0 +1,245 @@
+- A block can be **valid only if it includes an eligible agenda**. In other words, empty blocks are not allowed.
+- To decide which eligible agenda to finalize is the role of consensus, not the governance.
+- The consensus is a BFT algorithm based on a leader-and-round style.
+
+### What can be an agenda?
+
+- Transaction is only the unit of a **state transition**, not an individual item signed and broadcasted as in the case of a conventional blockchain.
+- An 'agenda' is defined as `(Proposer, Height, [Transaction])`.
+- Only a **single agenda** can be included in a block. This is for preventing complicated dependency problem.
+  - If multiple agendas are allowed, each agenda must be dependency-free; Otherwise, voters can't be sure how each agenda is ordered, or even whether it is included in the finalized block. This will cause a severe restriction of the possible agenda items.
+- Because `Height` is a part of an agenda,
+  every votes will be outdated and thus discarded if the block height progresses.
+
+## Chatting
+
+Again, Simperby is a blockchain engine that runs a **standalone, soverign, and self-hosted** organiziation.
+Thus, the communication channel for the organization should be so as well.
+The only way to implement such channel (instead of Discord) would be leveraging the existing P2P network.
+
+- Any member can broadcast its message, signed by their public key.
+- Each message contains a list of hashes of every previous messages that the signer has perceived, **forming a chain**.
+- Like PoW, the longest chat chain is considered as a candidate of the canonical one.
+- The consensus leader (block proposer) for the round plays a special role.
+  - the leader can broadcast a chat as well, but such chat is considered as a **semifinalized point**.
+  - A *faithful* leader would **frequently** insert its `ack` chat **on the (seemingly) longest chain**.
+  - A *good* leader must not commit a fork (semifinalizing two conflicting chains)
+- On top of the recent leader-semifinalized chat, other members continue chatting using the longest-chain rule until the next finalization point.
+- Once the governance reachs an eligible agenda, the leader must semifinalize the chat chain for the las time, and include the it in the block. If the block is finalized by the consensus then the chat log for the block is finalized as well.
+- A *good* leader must include the last-semifinalized chat chain which is the one that the other members would consider as the canonical one.
+
+### Why doesn't the leader just serve a chat server?
+
+The reason that the leader only 'semifinalizes' the chat chain, instead of running a full chat server, is that it is more censorship-resistant.
+
+- Compared to a full chat server, the longest-chain protocol has a
+  **weak** (becase of the lack of network synchrony) **consensus** (longest chain) of the canonical total chat ordering, among the members.
```

Comment:
```suggestion
  **weak** (because of the lack of network synchrony) **consensus** (longest chain) of the canonical total chat ordering, among the members.
```